lib/Driver/ToolChains.cpp

```diff
@@ -1210,6 +1210,10 @@ std::string toolchains::GenericUnix::getDefaultLinker() const {
   }
 }
 
+std::string toolchains::GenericUnix::getTarget() const {
```

Comment:
Can you name this something like "getOverridingTarget" or "getTargetForLinker"?

Also, does it make sense to fold this into "shouldSpecifyTargetTripleToLinker", now that we have that? Maybe there's just one "getTargetForLinker" that returns the current target by default, an override on Android, and no target on Cygwin (or wherever).